core/debug/src/debug_box.c

```diff
@@ -57,16 +60,22 @@ void debug_halt_error(THaltError reason)
     } else {
         /* Remember that debug_deprivilege_and_return() has been called once. */
         debugged_once_before = 1;
+        
+        /* Place the halt info on the interrupt stack. */
+        if (halt_info) {
+            g_debug_interrupt_sp[g_debug_box.box_id] -= sizeof(THaltInfo);
+            info = (void *)g_debug_interrupt_sp[g_debug_box.box_id];
+            memcpy(info, halt_info, sizeof(THaltInfo));
```

Comment:
This is okish. The alternative would be that the debug box reserves a `THaltInfo` in static memory and passes the pointer explicitly to uVisor either when registering the debug driver, or through the uVisor header input.
Note that memory consumption is the same, the additional memory on the stack has to be always considered.